feature-libs/product-configurator/rulebased/components/price/configurator-price.component.spec.ts

```diff
@@ -57,92 +50,129 @@ describe('ConfiguratorPriceComponent', () => {
     expect(component).toBeTruthy();
   });
 
-  describe('formula data is defined', () => {
-    it('should be defined: quantity equal to or greater than 1', () => {
-      component.formula = createTestData(5, 0, 0, true, true);
+  describe('Display quantity', () => {
+    it('should display a complete price formula', () => {
+      component.formula = createFormula(2, 10, 20);
       fixture.detectChanges();
 
-      CommonConfiguratorTestUtilsService.expectElementNotPresent(
-        expect,
-        htmlElem,
-        '.cx-price'
-      );
-      CommonConfiguratorTestUtilsService.expectElementNotPresent(
+      CommonConfiguratorTestUtilsService.expectElementPresent(
         expect,
         htmlElem,
         '.cx-quantity-price'
       );
-    });
-    it('should be defined: value price greater than zero', () => {
-      component.formula = {
-        quantity: 0,
-        price: {
-          currencyIso: '$',
-          formattedValue: '$10',
-          value: 10,
-        },
-        priceTotal: undefined,
-        isLightedUp: true,
-      };
-      fixture.detectChanges();
 
       CommonConfiguratorTestUtilsService.expectElementPresent(
         expect,
         htmlElem,
-        '.cx-price'
+        '.cx-price-total'
       );
+      const qty = component.formula.quantity;
+      const result =
+        component.formula.quantity +
+        'x(' +
+        component.formula.price?.formattedValue +
+        ')';
+      if (qty) {
+        expect(component.quantityWithPrice(qty.toString())).toEqual(result);
+      } else {
+        fail();
+      }
+      expect(component.priceTotal).toEqual(
+        '+' + component.formula.priceTotal?.formattedValue
+      );
```

Comment:
please do not use dynamic assertions. just expect a string literal or a const such as
`expect(component.priceTotal).toEqual('+$420.00')`
this way it is much easier to understand the test and its expectation. Additionally these dynamic expectations just duplicate the business logic they are supposed to test into the test itself.

Reply:
Don't agree but done!